### PR TITLE
BUG: Don't add debug files as targets when not building in debug mode

### LINKED ISSUE #10639: intro-install-plan.json does not match with intro-installed.json
**Describe the bug**

The files listed in the intro-install-plan.json do not match with the files installed (listed in intro-installed.json). 
This causes problems for meson-python, which uses intro-install-plan.json to find installed files.

In particular, on Windows, when building in release mode, pdb files will be listed in the install-plan but not installed, with a message like
```
File 'pandas\\_libs\\window\\indexers.cp38-win_amd64.pdb' not found, skipping
```

In addition, folders installed with install_subdir are not listed in the install-plan.

**To Reproduce**
Please include your `meson.build` files, preferably as a minimal toy example showing the issue.
You may need to create simple source code files (don't include private/proprietary code).

I found this problem while trying to port pandas to meson. Here is the code to reproduce.
```
git clone https://github.com/lithomas1/pandas.git@meson-poc
cd pandas
python -m build
```

It should be reproducible without pandas, though. I'll try to make a MRE when I have some time.
**Expected behavior**

files listed in intro-install-plan should match files in intro-installed.

**system parameters**
* Is this a [cross build](https://mesonbuild.com/Cross-compilation.html) or just a plain native build (for the same computer)?
native
* what operating system (e.g. MacOS Catalina, Windows 10, CentOS 8.0, Ubuntu 18.04, etc.)
Windows
* what Python version are you using e.g. 
3.8.0
* what `meson --version`
0.63.0
* what `ninja --version` if it's a Ninja build

### LINKED ISSUE #10775: Issues with python.extension_module() on Windows with MSVC
I would like to build a simple Python extension. My `meson.build` looks something like this:
```
py_mod = import('python')
py = py_mod.find_installation()
py_dep = py.dependency()

py.extension_module(
  'siphash24', 'siphash24.pyx', 'c-siphash.c',
  install: true,
  dependencies: py_dep,
)
```
the full code can be found here https://github.com/dnicolodi/python-siphash24

Everything works well on Linux and macOS however something is off on Windows when compiling with MSVC:
```
[4/4] Linking target siphash24.cp310-win_amd64.pyd
   Creating library siphash24.cp310-win_amd64.lib and object siphash24.cp310-win_amd64.exp
[...]
Installing siphash24.cp310-win_amd64.pyd to C:\Users\RUNNER~1\AppData\Local\Temp\build-via-sdist-g78k95j9\siphash24-1.0.0\.mesonpy-b3ixowso\install\hostedtoolcache\windows\Python\3.10.6\x64\Lib\site-packages
Installing siphash24.cp310-win_amd64.lib to C:\Users\RUNNER~1\AppData\Local\Temp\build-via-sdist-g78k95j9\siphash24-1.0.0\.mesonpy-b3ixowso\install\hostedtoolcache\windows\Python\3.10.6\x64\Lib\site-packages
File 'siphash24.cp310-win_amd64.pdb' not found, skipping
```
It seems that (what I think are) debug symbols are generated with the wrong filename extension.

This is on the Github Actions runners with the default MSVC installation. I'm not familiar with development on Windows thus I really don't know what may be going wrong.

### PR DESCRIPTION
**Pull request by lithomas1**
closes #10775
closes #10639

### CODE REVIEW

File: mesonbuild/build.py [hunk 1]
```diff
@@ -2146,7 +2146,14 @@ def determine_filenames(self):
                 prefix = ''
                 # Import library is called foo.lib
                 self.import_filename = self.vs_import_filename
-                create_debug_file = True
+                # Debug file is only created with debug buildtype
+                # (debug, debugoptimized)
+                create_debug_file = self.environment.coredata.get_option(OptionKey("buildtype")).startswith("debug")
```
Comment (dcbaker) [inline mesonbuild/build.py:2151]: you probably want to use the `debug` option, not the `buildtype`

File: mesonbuild/build.py [hunk 2]
```diff
@@ -2146,7 +2146,8 @@ def determine_filenames(self):
                 prefix = ''
                 # Import library is called foo.lib
                 self.import_filename = self.vs_import_filename
-                create_debug_file = True
+                # Debug files(.pdb) is only created with debug buildtype
+                create_debug_file = self.environment.coredata.get_option(OptionKey("debug"))
```
Comment (dcbaker) [inline mesonbuild/build.py:2150]: Shoukd this be set in tge rus case as well?

Comment (dcbaker) [reply mesonbuild/build.py:2150]: Actually, it lloks like we should juat have debug files as a possibility unless were using a gcc like compiler…

**Comment by lithomas1** [comment]
gentle ping @dcbaker @eli-schwartz for another review. Thanks.

**Comment by eli-schwartz** [comment]
Previously, we unconditionally set up debug files for MSVC-like compilers, and for rust, but never for GCC-like compilers (i.e. mingw).
Now we set it up in all 3 cases, but only for debug build types.

Is this correct that we do so for mingw too?

**Comment by lithomas1** [comment]
You're right, mingw doesn't generate a pdb file. Thanks for the catch.

**Comment by lithomas1** [comment]
gentle ping @eli-schwartz (if you're not too busy).

**Comment by eli-schwartz** [comment]
Thanks for reminding me! The holiday season just ended last night, and now I have lots more time but I do need to remember everything that needs my attention...

**Pull request merged**